src/main/java/cn/nukkit/inventory/ShapelessRecipe.java

```diff
@@ -19,10 +19,22 @@
 
     private final List<Item> ingredients = new ArrayList<>();
 
+    private int recipeProtocol = 130;
+
     public ShapelessRecipe(Item result) {
         this.output = result.clone();
     }
 
+    @Override
+    public boolean isCompatibleWith(int protocolVersion) {
```

Comment:
Srsly :/

Reply:
Yes, because 1.1 will crash if you send it recipes from 1.3

Reply:
1.2*

Reply:
Yes bro, but use one boolean function with true and false. Why true any time?